### PR TITLE
feat: bump otel to v0.104.0

### PR DESCRIPTION
**Pull request by davidcastai**

### CODE REVIEW

File: builder-config.yaml [hunk 1]
```diff
@@ -2,25 +2,25 @@ dist:
   name: castai-collector
   description: CAST AI OTel Collector that outputs Audit Logs to the console
   output_path: ./castai-collector
-  otelcol_version: 0.91.0
+  otelcol_version: 0.103.0
```
Comment (oskarwojciski) [inline builder-config.yaml:5]: Why not 0.104.0 as well?
```suggestion
  otelcol_version: 0.104.0
```

File: Makefile [hunk 2]
```diff
@@ -3,7 +3,8 @@ GOARCH := $(shell go env GOARCH)
 .PHONY: setup # Set up required tools (builder, mdatagen)
 setup:
 	go install go.opentelemetry.io/collector/cmd/builder@latest
-	go install github.com/open-telemetry/opentelemetry-collector-contrib/cmd/mdatagen@latest
+	chmod +x ./scripts/setup_mdatagen.sh
+	./scripts/setup_mdatagen.sh
```
Comment (oskarwojciski) [inline Makefile:7]: Why that change? `go install` seems to be working fine, we can also add some info to ensure that GOPATH is added to PATH variable:

```
@echo "\nNOTE: Ensure that" `go env GOPATH` "/bin is in your PATH env variable"
```

File: builder-config.yaml (hunk 1 above)
Comment (davidcastai) [reply builder-config.yaml]: missed this one, Thanks!

File: Makefile (hunk 2 above)
Comment (davidcastai) [reply Makefile]: removed script, will be proceeding `opentelemetry-collector-contrib` way

**Review by oskarwojciski — APPROVED** [review]

**Pull request merged**